### PR TITLE
Replaced version (unittest) with version (StdUnittest) to avoid unnecessary overhead when compiling with -unittest

### PR DESCRIPTION
**Pull request by alexandrumc**
Related to https://github.com/dlang/druntime/pull/2888

### CODE REVIEW

**Comment by CyberShadow** [comment]
Won't this break compiling user programs with `-unittest`? Meaning, due to `-unittest`, DMD will try to parse Phobos unittests too, which will now fail due to required definitions being versioned out?

**Comment by alexandrumc** [comment]
> Won't this break compiling user programs with `-unittest`? Meaning, due to `-unittest`, DMD will try to parse Phobos unittests too, which will now fail due to required definitions being versioned out?

No, because parsing doesn't have anything to do with valid definitions. Semantic analysis checks if the referred symbols are defined or not, but it is not done on the `unittest {}` blocks placed in libraries.

**Comment by CyberShadow** [comment]
> No, because parsing doesn't have anything to do with valid definitions.

Sorry for using the wrong term :)

> Semantic analysis checks if the referred symbols are defined or not, but it is not done on the `unittest {}` blocks placed in libraries.

Meaning, DMD will not look inside `unittest` blocks for files that are not on its command line, even with `-unittest`. I see.

Then, the question is, why not make `version(unittest)` behave the same way as `unittest` blocks? I.e. make the compiler ignore them if the file is not on its command line.

**Comment by alexandrumc** [comment]
> Then, the question is, why not make `version(unittest)` behave the same way as `unittest` blocks? I.e. make the compiler ignore them if the file is not on its command line.

Sure, a design decision must be made. However, we have to take into account the fact that `version {}` and `unittest {}` are different mechanisms, and ignoring version(unittest) might be only an unwanted particular case.

**Comment by alexandrumc** [comment]
@atilaneves

File: posix.mak [hunk 1]
```diff
@@ -150,7 +150,7 @@ ifdef ENABLE_COVERAGE
 override DFLAGS  += -cov
 endif
 
-UDFLAGS=-unittest
+UDFLAGS=-unittest -version=StdUnittest -version=CoreUnittest
```
Comment (MoonlightSentinel) [inline posix.mak]: Why include druntime's unittests with `CoreUnittest`?

Comment (alexandrumc) [reply posix.mak]: Phobos uses https://github.com/dlang/druntime/blob/9baf9735c923a9bbb72b8e4ee05f513ea0499bbc/src/core/time.d#L120-L123 to run its unittests. But that `version (unittest)` will change into `version (CoreUnittest)` (https://github.com/dlang/druntime/pull/2902), so the flag has to be there.

Comment (MoonlightSentinel) [reply posix.mak]: Right, -`unittest` cascades into druntime. Thanks for the explaination.

File: std/conv.d [hunk 2]
```diff
@@ -5687,10 +5688,6 @@ pure nothrow @safe /* @nogc */ unittest
 
 @system unittest
 {
-    int var = 6;
```
Comment (atilaneves) [inline std/conv.d]: Why are these lines deleted?

Comment (alexandrumc) [reply std/conv.d]: Because I moved the content of a `version (unittest)` inside a `unittest {}`, so I had to move those lines too. They are now at line 5009.

File: posix.mak (hunk 1 above)
Comment (PetarKirov) [reply posix.mak]: Should now be not necessary with my changes to https://github.com/dlang/druntime/pull/2902.

Comment (PetarKirov) [reply posix.mak]: Addressed in https://github.com/dlang/phobos/pull/7356.
